File: src/src_core/fmg/core/mosaic/draw/PaintMosaicContext.java
package fmg.swing.draw.mosaic;

import java.awt.Color;

import javax.swing.ImageIcon;
import javax.swing.JComponent;

import fmg.swing.Cast;
import fmg.swing.draw.GraphicContext;

public class MosaicGraphicContext extends GraphicContext {
   private Color     colorBk;
   private ImageIcon imgBckgrnd;

   public MosaicGraphicContext(JComponent owner) {
      super(owner, false);
   }

   public Color getColorBk() {
      if (colorBk == null) {
         fmg.common.Color clr = GraphicContext.getDefaultBackgroundFillColor();
         setColorBk(Cast.toColor(clr.darker(0.4)));
      }
      return colorBk;
   }

   public void setColorBk(Color colorBk) {
      Color old = this.colorBk;
      if (colorBk.equals(old))
         return;
      this.colorBk = colorBk;
      onPropertyChanged(old, colorBk, "ColorBk");
   }

   public ImageIcon getImgBckgrnd() {
      return imgBckgrnd;
   }

   public void setImgBckgrnd(ImageIcon imgBckgrnd) {
      Object old = this.imgBckgrnd;
      if (old == imgBckgrnd) // references compare 
         return;
      this.imgBckgrnd = imgBckgrnd;
      onPropertyChanged(old, imgBckgrnd, "ImgBckgrnd");
   }

}

File: src/src_core/fmg/core/mosaic/draw/PaintMosaicContext.cs
using fmg.common;

namespace fmg.core.mosaic.draw {

   public class PaintMosaicContext<TImage> : PaintCellContext<TImage>
      where TImage : class
   {
      private Color _colorBk;
      private TImage _imgBckgrnd;

      public PaintMosaicContext(bool iconicMode) :
         base(iconicMode)
      {
         _colorBk = DefaultBackgroundWindowColor.Darker(0.4);
      }

      public Color ColorBk {
         get { return _colorBk; }
         set { SetProperty(ref _colorBk, value); }
      }

      public TImage ImgBckgrnd {
         get { return _imgBckgrnd; }
         set { SetProperty(ref _imgBckgrnd, value); }
      }

   }

}
